### ISSUE TITLE
`IPython.display.Audio` creates a widget to play the sound, but the sound isn't played after clicking the play button.

### ISSUE THREAD

**Issue by mmagnuski** [jupyter/notebook#4321]
Using this code snippet in notebook:
```python
from IPython.display import Audio
import numpy as np

signal = np.random.random(750)
Audio(signal, rate=250)
```
gives the player widget, but clicking the play sound:
* doesn't play any sound
* the time slider doesn't move
* although the sound button changes the graphic to pause

![audio_bug_example](https://user-images.githubusercontent.com/8452354/50769631-dc266e80-1284-11e9-9ea3-e51a66125004.gif)

## Versions:
```
Windows 10

# Name                    Version                   Build  Channel
notebook                  5.0.0            py36hd9fbf6f_2
spyder-notebook           0.1.3                    py36_0    spyder-ide
jupyter                   1.0.0            py36h422fd7e_2
jupyter_client            5.1.0            py36h9902a9a_0
jupyter_console           5.2.0            py36h6d89b47_1
jupyter_core              4.3.0            py36h511e818_0
jupyterlab                0.30.1                   py36_0    conda-forge
jupyterlab_launcher       0.6.0                    py36_0    conda-forge
ipywidgets                7.0.0            py36h2e74ada_0
```
but I had the same problem 10 minutes ago on another Windows 10 PC, where I updated multiple jupyter-related packages.

**Comment by mmagnuski** [comment]
However, clicking on `...` and saving the `.vaw` file and then opening it in Windows does play the file.

**Comment by takluyver** [comment]
Which browser are you using, and can you try in another one? The display and playback of the audio is done by your browser.

**Comment by mmagnuski** [comment]
I was using chrome on both computers. Both should have the current chrome version: `71.0.3578.98` (one has this version for sure, because I just checked, the other computer is unavailable at the moment). 
Sure, I can try another browser - what browser is recommended for audio playback?

**Comment by takluyver** [comment]
I don't know if any are particularly recommended, just that if it works in one browser and not another, we'll definitely blame the browser. ;-)

Firefox is available for all platforms and a supported browser for Jupyter, so that's an obvious one to try.

**Comment by MartyMacGyver** [comment]
Still a problem in Chrome 72.0.3626.109 (Official Build) (64-bit). I'm binding autoplay sounds to an array of buttons from ipywidgets. Clicking said buttons used to work but no longer does.

Here's some notebook test code (tried in 3.6 and 3.7). This is directly from
https://ipython-books.github.io/117-creating-a-sound-synthesizer-in-the-notebook/

```
import numpy as np
import matplotlib.pyplot as plt
from IPython.display import (
    Audio, display, clear_output)
from ipywidgets import widgets
from functools import partial
%matplotlib inline
```

```
rate = 16000.
duration = .25
t = np.linspace(
    0., duration, int(rate * duration))

def synth(f):
    x = np.sin(f * 2. * np.pi * t)
    display(Audio(x, rate=rate, autoplay=True))

notes = 'C,C#,D,D#,E,F,F#,G,G#,A,A#,B,C'.split(',')
freqs = 440. * 2**(np.arange(3, 3 + len(notes)) / 12.)
notes = list(zip(notes, freqs))

layout = widgets.Layout(
    width='30px', height='60px',
    border='1px solid black')

buttons = []
for note, f in notes:
    button = widgets.Button(
        description=note, layout=layout)

    def on_button_clicked(f, b):
        with widgets.Output():
            synth(f)

    button.on_click(partial(on_button_clicked, f))
    buttons.append(button)

widgets.Box(children=buttons)

#synth(440)
```

It all works in Firefox, but it then has the added feature of playing *at once* all the sounds loaded by all the button widgets when the page is first loaded. "Works with new, weird side-effects" is not acceptable.

Delving deeper, this MIGHT stem from recent changes to Chrome to mute spammy sounds (autoplay videos, etc.). I did "opt in" for sound from this page via site settings but that didn't help.
https://developers.google.com/web/updates/2017/09/autoplay-policy-changes

If `synth(f)` is NOT enclosed in `with widgets.Output():` this works even on Chrome - while generating a very undesirable "play" widget for Every. Single. Button-press. That clause seems to be at the heart of the problem though why it affects Chrome is a bit of a mystery to me - is it because this mimics a "hidden" element that would otherwise be used to inject spam or ad media? Firefox seems to ignore this risk... but Firefox also allows all sorts of spam autoplay behavior as well by default so again, not helpful.

**Comment by MartyMacGyver** [comment]
Also some discussion here:

https://stackoverflow.com/questions/50490304/how-to-make-audio-autoplay-on-chrome

**Comment by MartyMacGyver** [comment]
_....Time passed..._

@mmagnuski, what you experienced originally is actually a different problem from mine.

Given `Audio(signal, rate=rate )`, `rate` needs to be a reasonable audio sampling rate, as described below:

```
from IPython.display import Audio, display
import numpy as np

frequency = 440
duration = 0.5  # How long to play the sound in seconds

rate = 8000  # The sampling rate in Hz (should be twice the Nyquist frequency)
# Basically, if you want to sample sounds up to frequency f,
# your sampling rate should be >= 2*f
# e.g., An 8kHz sampling rate works for frequencies up to 4kHz
# It's also limited by the browser/audio subsystem - 3kHz seems to be the min
# for Chrome + Windows - it's barely audible for the noise example

print('Play random hoise')
signal_noise = np.random.random(int(rate * duration))
display(Audio(signal_noise, rate=rate))

print('Play a tone of {} Hz'.format(frequency))
signal_tone = np.sin(frequency * 2. * np.pi * np.linspace(0., duration, int(rate * duration)))
display(Audio(signal_tone, rate=rate))
```

You'll notice that as you drop `rate` from 8kHz to 3kHz the noise sound gets progressively lower. Even if it somehow worked (rather than just stopping entirely at some low value), random noise (and almost any other sound) at a sample rate of 250Hz would be virtually inaudible.

**Comment by mmagnuski** [comment]
Yes, @MartyMacGyver, sorry if my silly example was confusing. I originally used mixtures of sinusoids and eeg signal with the same behavior. What I wanted to demonstrate in the video was not lack of sound (since that I cannot capture in a gif) but the behavior of the slider (lack of movement) and I might have simplified the example too much.

**Comment by MartyMacGyver** [comment]
I understand - with any rate under 3000 I see the same (and no slider movement either). The rate has to be high enough for that widget to work.
